### PR TITLE
[PIL-802][DG]Adding contact Details & Print on final CYA page

### PR DESCRIPTION
**Pull request by dhirenganvir**

### CODE REVIEW

**Review by neelu-dwivedi — COMMENTED** [review]
Can you add this scneario in any of the existing feature file for RFM journey rather than creating a new for 1 scenario?

**Review by neelu-dwivedi — APPROVED** [review]

File: src/test/resources/features/RFMFinalReviewCYAPage.feature [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    And I should see row 3 value 76543210
+    And I should see row 4 value 1234567890
+    When I click change link for RFM Company
+    Then I should be on RFM registered in UK page
+    And I click on Continue button
+    Then I should navigate to RFM UK based entity type page
+    When I select back link
+    Then I should be on RFM registered in UK page
+
+
```
Comment (srikanthhmrc) [inline src/test/resources/features/RFMFinalReviewCYAPage.feature:128]: can we remove extra spaces

**Review by srikanthhmrc — CHANGES_REQUESTED** [review]

File: src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/CommonFunctions.scala [hunk 2]
```diff
@@ -138,18 +138,19 @@ trait CommonFunctions
...
       case "RFM GRS Registration Mismatch Error Page"   => RFMGRSRegistrationMismatchErrorPage
-      case "RFM Second Contact Address Page"            => RFMSecondContactAddressPage
-      case "RFM Contact Details Final CYA Page"         => RFMContactDetailsFinalCYAPage
-      case "RFM Contact Address Page"                   =>RFMContactAddressPage
-      case "RFM Change Contact Details CYA Page"        =>RFMChangeContactDetailsCYAPage
-      case "ASA Pillar2 Input Page"                     => ASAPillar2InputPage
-      case "RFM No record Match Error Page"             => RFMNoMatchErrorPage
+      case "RFM Second Contact Address Page"      => RFMSecondContactAddressPage
+      case "RFM Final Review Page"                => RFMFinalReviewCYAPage
+      case "RFM Second Contact Address Page"      => RFMSecondContactAddressPage
```
Comment (srikanthhmrc) [inline src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/CommonFunctions.scala:149]: remove the duplicate step

File: src/test/resources/features/RFMFinalReviewCYAPage.feature [hunk 3]
```diff
@@ -0,0 +1,142 @@
+@tests
```
Comment (srikanthhmrc) [inline src/test/resources/features/RFMFinalReviewCYAPage.feature:1]: can we put this back in RFMCYAPage.feature file

File: src/test/resources/features/RFMFinalReviewCYAPage.feature [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    And I select option No and continue to next
+    Then I should be on RFM Second Contact Address Page
+    When I enter Address Line 1 as RFM Address Line 1
+    And I enter City as RFM City
+    And I enter Postal Code as EH5 5WY
+    And I enter Country as United Kingdom
+    And I click on Country selected
+    When I click on Continue button
+    Then I should navigate to RFM Final Review Page
+    And The Heading should be Review and submit
```
Comment (srikanthhmrc) [inline src/test/resources/features/RFMFinalReviewCYAPage.feature:33]: need to remove content verification steps and make sure that these are covered in unit tests as per HMRC test leads suggestion

**Review by srikanthhmrc — CHANGES_REQUESTED** [review]

**Review by annemonica5 — APPROVED** [review]

**Pull request merged**